src/GitHubTools/Functions/Public/Update-AzurePipelineDependencies.ps1

```diff
@@ -0,0 +1,118 @@
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseSingularNouns", "")]
+    param(
+        [Parameter(Mandatory = $false)]
+        [String]$GitHubOrganisation = "SkillsFundingAgency",
+        [Parameter(Mandatory = $false)]
+        [String]$RepositoryPrefix = "das-"
+    )
+
+    $Repos = Get-GitHubRepo -GitHubOrg $GitHubOrganisation | Where-Object { $_.name -match "^$RepositoryPrefix.+" -and $_.isArchived -eq $False }
+
+    $Repos | ForEach-Object {
+        $Repo = $_
+
+        Write-Host "Scanning $($Repo.name)"
+
+        $PipelineYml = Get-GitHubRepoFileContent -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name -FilePath "azure-pipelines.yml"
+
+        if (!$PipelineYml) { return }
+
+        $Pipeline = ConvertFrom-Yaml -Yaml $PipelineYml.Content
+
+        $UpdatePipelineYml = $false
+        $Pipeline.resources.repositories | Where-Object { $_.type -eq "github" } | ForEach-Object {
+            if ($_.ref -like "refs/tags/*") {
+                $TagName = $_.ref.replace("refs/tags/", "")
+                $DependencyOrg = $_.name.Split("/")[0]
+                $DependencyRepo = $_.name.Split("/")[1]
+                $Releases = Get-GitHubRepoRelease -GitHubOrganisation $DependencyOrg -RepositoryName $DependencyRepo
+                $OutOfDate = $Releases[0].tag_name -ne $TagName
+
+                if ($OutOfDate) {
+                    Write-Warning "$($_.name) dependency: $TagName should be $($Releases[0].tag_name): updating"
+                    $PipelineYml.Content = $PipelineYml.Content.Replace($TagName, $Releases[0].tag_name)
+                    $UpdatePipelineYml = $true
+                }
+                else {
+                    Write-Host "$($_.name) dependency up to date" -ForegroundColor Green
+                }
+            }
+            else {
+                Write-Warning "No tag reference for: $($_.name) - $($_.ref)"
+                return
+            }
+        }
+
+        $PipelineYml.Content = $PipelineYml.Content.Replace("      - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks", "")
+
+        $CodeBuildYml = Get-GitHubRepoFileContent -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name -FilePath "pipeline-templates/job/code-build.yml"
+
+        $UpdateCodeBuildYml = $false
+        if ($CodeBuildYml) {
+            if ($CodeBuildYml.Content.Contains("  - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks")) {
+                $CodeBuildYml.Content = $CodeBuildYml.Content.Replace("  - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks", "")
+                $UpdateCodeBuildYml = $true
+            }
```

Comment:
Not sure how I feel about putting this in master. It's a very specific one time run use case?

Reply:
Even though it's specific, as we're likely to need to run the function on the repositories I think OK to have in master.

Then we'll remove when template step is no longer in any repos.

Reply:
Removed this one time run use case, will run off a hot-fix branch instead